R/wwinference.R

```diff
@@ -41,31 +41,102 @@
 #' @param compiled_model The pre-compiled model as defined using
 #' `compile_model()`
 #'
-#' @return A nested list of the following items, intended to allow the user to
-#' quickly and easily plot results from their inference, while also being able
-#' to have the full user functionality of running the model themselves in stan
-#' by providing the raw model object and diagnostics. If the model runs, this
+#' @return An object of the `ww_inference_fit` class containing the following
+#' items that are intended to be passed to downstream functions to do things
+#' like extract posterior draws, get diangostic behavior, and plot results
+#' (for example). If the model runs, this
 #' function will return:
-#' `draws_df`: A tibble containing the full set of posterior draws of the
-#' estimated, nowcasted, and forecasted: counts, site-level wastewater
-#' concentrations, "global"(e.g. state) R(t) estimate, and the  "local" (site +
-#' the one auxiliary subpopulation) R(t) estimates. In the instance where there
-#' are observations, the data will be joined to each draw of the predicted
-#' observation to facilitate plotting.
-#' `raw_fit_obj`: The CmdStan object that is returned from the call to
+#' `fit`: The CmdStan object that is returned from the call to
 #' `cmdstanr::sample()`. Can be used to access draws, summary, diagnostics, etc.
-#' `date_time_spine`: Mapping from time in stan to dates
-#' `lab_site_spine`: Mapping from lab_site_index in stan to lab and site
-#' `subpop_spine`: Mapping from site index in stan to site
+#' `input_data`: a list containing the input `ww_data` and the input
+#' `count_data` used in the model.
+#' `stan_data_args`: a list containing the inputs passed directly to the
+#' stan model
+#' `mcmc_options`: a list of the MCMC specifications passed to stan
```

Comment:
@seabbs Curious your thoughts on these outputs of the `wwinference()` wrapper function, as these would be what I would hope would stay stable even as we continue to develop on this.